treeherder/etl/job_loader.py

```diff
@@ -59,6 +60,38 @@ def process_job_list(self, all_jobs_list):
 
                 jobs_model.store_job_data(storeable_job_list)
 
+    def clean_revision(self, pulse_job, jobs_model):
+        # It is possible there will be either a revision or a revision_hash
+        # At some point we will ONLY get revisions and no longer receive
+        # revision_hashes and then this check can be removed.
+        revision = pulse_job["origin"].get("revision", None)
+        if revision:
+            # check the revision for this job has an existing resultset
+            # If it doesn't, then except out so that the celery task will
+            # retry till it DOES exist.
+            if not jobs_model.get_resultset_top_revision_lookup([revision]):
+                raise MissingResultsetException(
+                    "No resultset found for revision {}".format(revision))
+
+        else:
+            # This will also raise a ValueError if the resultset for the
+            # revision_hash is not found.
+            revision = jobs_model.get_revision_from_revision_hash(
+                pulse_job["origin"]["revision_hash"]
+            )
+            logger.warning(
+                "Pulse job had revision_hash instead of revision: {}:{}".format(
+                    pulse_job["origin"]["project"],
+                    pulse_job["origin"]["revision_hash"]
+                ))
+            params = {
+                "project": pulse_job["origin"]["project"],
+                "revision_hash": pulse_job["origin"]["revision_hash"]
+            }
+            newrelic.agent.record_custom_event("revision_hash_usage", params=params)
```

Comment:
Yeah, it's older branches.  There were just a batch of them coming from I think Aurora or something a week or so ago (there's a saved query in Insights for it).  However, After this quarter all the code that uses revision_hashes MAY actually be out.  I was planning to follow up with garndt on this early next quarter to check our status.